### PR TITLE
Load the main bundle's routing file form app's routing.yml file

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes the problem described by @weaverryan in [this comment](https://github.com/sensiolabs/SensioGeneratorBundle/pull/377#issuecomment-124860772).

The error which will be usually displayed:

generator_1

The error when something really bad happens:

generator_2

Don't look at the style of the errors because I'll change everything soon. But please, review the actual content of the error messages. Thanks.

### CODE REVIEW

File: Command/GenerateDoctrineCrudCommand.php [hunk 1]
```diff
@@ -261,6 +263,37 @@ protected function updateRouting(QuestionHelper $questionHelper, InputInterface
                 '',
             );
         }
+
+        // second, import the bundle's routing.yml file from the application's routing.yml file
+        $routing = new RoutingManipulator($this->getContainer()->getParameter('kernel.root_dir').'/config/routing.yml');
+        try {
+            $ret = $auto ? $routing->addResource($bundle->getName(), $format) : false;
```
Comment (weaverryan) [inline Command/GenerateDoctrineCrudCommand.php]: the `$format` here should always be `yml`, correct? Because we want to import `@YourBundle/Resources/config/routing.yml`?

File: Command/GenerateDoctrineCrudCommand.php [hunk 2]
```diff
@@ -261,6 +263,37 @@ protected function updateRouting(QuestionHelper $questionHelper, InputInterface
...
+        } catch (\RuntimeException $e) {
+            // the bundle is already imported form app's routing.yml file
+            $errorMessage = sprintf(
+                "\n\n[ERROR] The bundle's \"Resources/config/routing.yml\" file cannot be imported\n".
+                "from \"app/config/routing.yml\" because the \"%s\" bundle is\n".
+                "already imported. Make sure you are not using two different\n".
+                "configuration/routing formats in the same bundle because it won't work.\n",
+                $bundle->getName()
+            );
```
Comment (weaverryan) [inline Command/GenerateDoctrineCrudCommand.php]: if I already have the import to `@AppBundle/Resources/config/routing.yml` in `app/config/routing.yml`, will this error be shown?

**Comment by weaverryan** [comment]
@javiereguiluz it feels like we need a battery of functional tests for this bundle to make sure the edge-cases are taken care of in a decent way. There's a lot of manual testing currently.

File: Command/GenerateDoctrineCrudCommand.php (hunk 2 above)
Comment (javiereguiluz) [inline Command/GenerateDoctrineCrudCommand.php]: Yes, because the check made by RoutingManipulator is very simple: https://github.com/sensiolabs/SensioGeneratorBundle/blob/master/Manipulator/RoutingManipulator.php#L55-L60

**Comment by javiereguiluz** [comment]
@weaverryan I agree completely. And we'd also need to refactor the entire bundle because most of its code is more complex than it should be.

File: Command/GenerateDoctrineCrudCommand.php (hunk 1 above)
Comment (javiereguiluz) [inline Command/GenerateDoctrineCrudCommand.php]: Good catch! Fixed. Thanks.

**Comment by fabpot** [comment]
What's the status of this PR?

**Comment by fabpot** [comment]
Thank you @javiereguiluz.
